packages/angular/build/src/tools/angular/transformers/lazy-routes-transformer.ts

```diff
@@ -0,0 +1,212 @@
+  compilerHost: ts.CompilerHost,
+): ts.TransformerFactory<ts.SourceFile> {
+  const moduleResolutionCache = compilerHost.getModuleResolutionCache?.();
+  assert(
+    typeof compilerOptions.basePath === 'string',
+    'compilerOptions.basePath should be a string.',
+  );
+  const basePath = compilerOptions.basePath;
+
+  return (context: ts.TransformationContext) => {
+    const factory = context.factory;
+
+    const visitor = (node: ts.Node): ts.Node => {
+      if (!ts.isObjectLiteralExpression(node)) {
+        // Not an object literal, so skip it.
+        return ts.visitEachChild(node, visitor, context);
+      }
+
+      const loadFunction = getLoadComponentOrChildrenProperty(node)?.initializer;
+      // Check if the initializer is an arrow function or a function expression
+      if (
+        !loadFunction ||
+        (!ts.isArrowFunction(loadFunction) && !ts.isFunctionExpression(loadFunction))
+      ) {
+        return ts.visitEachChild(node, visitor, context);
+      }
+
+      let callExpression: ts.CallExpression | undefined;
+      if (ts.isArrowFunction(loadFunction)) {
+        // For arrow functions, the body can be a call expression directly
+        if (ts.isCallExpression(loadFunction.body)) {
+          callExpression = loadFunction.body;
+        }
+      } else if (ts.isFunctionExpression(loadFunction)) {
+        // For function expressions, check if the body has a return statement with a call expression
+        const returnStatementExpression = loadFunction.body.statements.find(
+          ts.isReturnStatement,
+        )?.expression;
+
+        if (returnStatementExpression && ts.isCallExpression(returnStatementExpression)) {
+          callExpression = returnStatementExpression;
+        }
+      }
+
+      if (!callExpression) {
+        return ts.visitEachChild(node, visitor, context);
+      }
+
+      // Optionally check for the 'then' property access expression
+      const expression = callExpression.expression;
+      if (
+        !ts.isCallExpression(expression) &&
+        ts.isPropertyAccessExpression(expression) &&
+        expression.name.text !== 'then'
+      ) {
+        return ts.visitEachChild(node, visitor, context);
+      }
+
+      const importExpression = ts.isPropertyAccessExpression(expression)
+        ? expression.expression // Navigate to the underlying expression for 'then'
+        : callExpression;
+
+      // Ensure the underlying expression is an import call
+      if (
+        !ts.isCallExpression(importExpression) ||
+        importExpression.expression.kind !== ts.SyntaxKind.ImportKeyword
+      ) {
+        return ts.visitEachChild(node, visitor, context);
+      }
+
+      // Check if the argument to the import call is a string literal
+      const callExpressionArgument = importExpression.arguments[0];
+      if (
+        !ts.isStringLiteral(callExpressionArgument) &&
+        !ts.isNoSubstitutionTemplateLiteral(callExpressionArgument)
+      ) {
```

Comment:
Nit: Could we use [`ts.isStringLiteralLike`](https://github.com/microsoft/TypeScript/blob/3d2b8f33d4ff86ec351ba93a557d37720962c921/src/compiler/utilitiesPublic.ts#L2622-L2624) here?